shepherd_bms/src/main.cpp

```diff
@@ -111,44 +113,130 @@ void shepherdMain()
 	//Create a dynamically allocated structure
 	//@note this will move to a specialized container with a list of these structs
 	AccumulatorData_t *accData = new AccumulatorData_t;
+	
 
 	//Collect all the segment data needed to perform analysis
 	//Not state specific
 	segment.retrieveSegmentData(accData->chipData);
+
+	int16_t current = compute.getPackCurrent();
 	Serial.print("Current: ");
-	Serial.println(compute.getPackCurrent());
+	Serial.println(current);
 	//compute.getTSGLV();
 	//etc
 
 	//Perform calculations on the data in the frame
 	//Some calculations might be state dependent
 	calcCellTemps(accData);
 	calcPackTemps(accData);
+	calcPackVoltageStats(accData);
 	Serial.print("Min, Max, Avg Temps: ");
 	Serial.print(accData->minTemp.val);
 	Serial.print(",  ");
 	Serial.print(accData->maxTemp.val);
 	Serial.print(",  ");
 	Serial.println(accData->avgTemp);
+	Serial.print("Min, Max, Avg, Delta Voltages: ");
+	Serial.print(accData->minVoltage.val);
+	Serial.print(",  ");
+	Serial.print(accData->maxVoltage.val);
+	Serial.print(",  ");
+	Serial.print(accData->avgVoltage);
+	Serial.print(",  ");
+	Serial.println(accData->deltVoltage);
 	calcCellResistances(accData);
 	calcDCL(accData);
-	calcCCL(accData);
+	calcContDCL(accData);
+	calcContCCL(accData);
 	Serial.print("DCL: ");
 	Serial.println(accData->dischargeLimit);
 
-	Serial.println("Cell Resistances:");
+	Serial.print("CCL: ");
+	Serial.println(accData->chargeLimit);
+
+	/*
+	Serial.println("Cell Temps:");
 	for(uint8_t c = 0; c < NUM_CHIPS; c++)
     {
-        for(uint8_t cell = 0; cell < NUM_CELLS_PER_CHIP; cell++)
+        for(uint8_t cell = 17; cell < 28; cell++)
         {
-			Serial.print(accData->chipData[c].cellResistance[cell]);
+			Serial.print(accData->chipData[c].thermistorReading[cell]);
 			Serial.print("\t");
 		}
 		Serial.println();
 	}
 
-	//Send out what needs to happen now (depends on state)
-	//compute.sendMCMsg(CCL, DCL);
+	Serial.println("Cell Temps Avg:");
+	for(uint8_t c = 0; c < NUM_CHIPS; c++)
+    {
+        for(uint8_t cell = 17; cell < 28; cell++)
+        {
+			Serial.print(accData->chipData[c].thermistorValue[cell]);
+			Serial.print("\t");
+		}
+		Serial.println();
+	}*/
+
+	uint16_t overVoltCount = 0;
+
+	// ACTIVE/NORMAL STATE
+	if (bmsFault == FAULTS_CLEAR) {
+		compute.sendMCMsg(0, accData->dischargeLimit);
+
+		// Check for fuckies
+		if (current > accData->contDCL) {
+			bmsFault = DISCHARGE_LIMIT_ENFORCEMENT_FAULT;
```

Comment:
You shouldn't do it like this, since we want to have multiple faults at once, you should either do an ```|= FAULT_CODE``` to set it high or a ```&=~(FAULT_CODE)``` to set it low